src/distribution_spatial.cpp

```diff
@@ -180,6 +182,71 @@ Position SphericalIndependent::sample(uint64_t* seed) const
   return {x, y, z};
 }
 
+//==============================================================================
+// MeshSpatial implementation
+//==============================================================================
+
+MeshSpatial::MeshSpatial(pugi::xml_node node)
+{
+  // No in-tet distributions implemented, could include distributions for the barycentric coords
+  // Read in unstructured mesh from mesh_id value
+  int32_t mesh_id = std::stoi(get_node_value(node, "mesh_id"));
+  // Get pointer to spatial distribution
+  mesh_idx_ = model::mesh_map.at(mesh_id);
+
+  // Check whether mesh pointer points to a mesh
+  mesh_ptr_ = dynamic_cast<Mesh*>(model::meshes[mesh_idx_].get());
+  if (!mesh_ptr_) {fatal_error("Mesh passed to spatial distribution is not a mesh object"); }
```

Comment:
We shouldn't. Gonna pull these lines. Thanks!

Reply:
I believe this line is left over from when we used it to check whether the mesh pointer was of type unstructured mesh. Since we added functionality/error messages for all types of meshes it became redundant.